src/redisraft.c

```diff
@@ -731,6 +715,10 @@ static void handleRedisCommand(RedisRaftCtx *rr,
         return;
     }
 
+    /* update cmds array with the client's saved "asking" state */
+    cmds->asking = getAskingState(rr, ctx);
+    setAskingState(rr, ctx, false);
+
```

Comment:
I see we added this line to handle multi case: [link](https://github.com/RedisLabs/redisraft/blame/bef32837e3209df7111a6173040f9071d20b1f1e/src/serialization.c#L52)

So, we copy "asking" state but looks like we override it later. 

ASKING sets client's `asking` =  `true`
MULTI sets `multiState->cmds->asking` = `true` 
GET X sets `multiState->cmds->asking` = `false`


Probably, that line above can be 
```c
target->asking |= source->asking;
```
Once we set it, we don't want to unset it. 
In future, we can improve this logic probably.

----
Btw, I didn't test but IIUC, to match Redis' behavior, we should do in this order:

```c
static void handleRedisCommand(RedisRaftCtx *rr, RedisModuleCtx *ctx, 
                               RaftRedisCommandArray *cmds)
{
    /* update cmds array with the client's saved "asking" state */
    cmds->asking = getAskingState(rr, ctx);
    setAskingState(rr, ctx, false);

    /* Check if MULTI/EXEC bundling is required. */
    if (MultiHandleCommand(rr, ctx, cmds)) {
        return;
    }

    /* Handle intercepted commands. We do this also on non-leader nodes or if we
     * don't have a leader, so it's up to the commands to check these conditions
     * if they have to.
     */
    if (handleInterceptedCommands(rr, ctx, cmds)) {
        return;
    }

    handleRedisCommandAppend(rr, ctx, cmds);
}
```

Redis handles "multi" before other commands, so, I guess we can do in this order if it doesn't cause any other problems.